pkg/middleware/validation.go

```diff
@@ -45,31 +47,60 @@ func NewValidationMiddleware(validatorHook string) mux.MiddlewareFunc {
 	}
 }
 
+func maybeLogValidationReason(valid ValidationResponse, r *http.Request, mod string, version string) {
+	message := valid.ValidationMessage
+	if message.Reason != "" {
+		entry := log.EntryFromContext(r.Context())
+		entry.Warnf("Module %s:%s Reason: %s, Description: %s", mod, version, message.Reason, message.Description)
+	}
+}
+
 type validationParams struct {
 	Module  string
 	Version string
 }
 
-func validate(hook, mod, ver string) (bool, error) {
+type ValidationMessage struct {
+	Reason      string
+	Description string
+}
+
+type ValidationResponse struct {
+	Success           bool
```

Comment:
what is this boolean used for? Conventionally, status OK means Success=true while status FORBIDDEN means Success=false. Does this serve another purpose?

Reply:
It is essentially still achieving the same result. I was just making it more explicit on the meaning that forbidden == success.
Here I am setting `Success` based on the response code https://github.com/gomods/athens/pull/1517/files#diff-7522a4893602f93e19a3c36e780166e2R101 . I can revert this if you would prefer to pass around the status code instead.

Reply:
I have changed the success to carry the status code instead https://github.com/gomods/athens/pull/1517/commits/933d4430dadc1cfd1b8a34d85162930e503434b3